packages/trader/src/AppV2/app.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import type { TWebSocket } from 'Types';
+import initStore from 'App/init-store';
+import 'Sass/app.scss';
+import type { TCoreStores } from '@deriv/stores/types';
+import TraderProviders from '../trader-providers';
+import BottomNav from './Components/BottomNav';
+import Trade from './Containers/Trade';
+import Markets from './Containers/Markets';
+import Positions from './Containers/Positions';
+import Menu from './Containers/Menu';
+
+type Apptypes = {
+    passthrough: {
+        root_store: TCoreStores;
+        WS: TWebSocket;
+    };
+};
+
+const App = ({ passthrough }: Apptypes) => {
+    const root_store = initStore(passthrough.root_store, passthrough.WS);
+    React.useEffect(() => {
+        return () => root_store.ui.setPromptHandler(false);
+    }, [root_store]);
```

Comment:
Can it track changes in such huge object as `root_store`? Please, ignore if irrelevant

Reply:
i copied pasta app.tsx in AppV1. Not entirely sure what its doing tbh.

Reply:
@kate-deriv React.useEffect will just track the reference of the object and it won't track the nested props. So, this code is safe in terms of performance.